pkg/cloudfabric-controller/controllerframework/controller_framework.go

```diff
@@ -199,24 +169,28 @@ func (c *ControllerBase) WatchInstanceUpdate(stopCh <-chan struct{}) {
 				klog.Error("Unexpected controller instance update message")
 				return
 			}
-			klog.Infof("Got controller instance update massage. Updated Controller Type %s, current controller instance type %s, key %d", updatedType, c.controllerType, c.controllerKey)
-			if updatedType != c.controllerType {
-				continue
-			}
-
-			klog.V(4).Infof("Start updating controller instance %s", c.controllerType)
-			controllerInstances, err := listControllerInstancesByType(c.controllerType)
-			if err != nil {
-				// TODO - add retry
-				klog.Errorf("Error getting controller instances for %s. Error %v", c.controllerType, err)
-				continue
-			}
-			c.updateCachedControllerInstances(controllerInstances)
-			klog.V(4).Infof("Done updating controller instance %s", c.controllerType)
+			c.instanceUpdateProcess(updatedType.(string))
 		}
 	}
 }
 
+func (c *ControllerBase) instanceUpdateProcess(updatedType string) {
+	klog.Infof("Got controller instance update massage. Updated Controller Type %s, current controller instance type %s, key %d", updatedType, c.controllerType, c.controllerKey)
+	if updatedType != c.controllerType {
+		return
+	}
+
+	klog.V(4).Infof("Start updating controller instance %s", c.controllerType)
+	controllerInstances, err := listControllerInstancesByType(c.controllerType)
+	if err != nil {
+		// TODO - add retry
+		klog.Errorf("Error getting controller instances for %s. Error %v", c.controllerType, err)
```

Comment:
Nit: the log message should be " Error getting controller instances for type %s". It looks like %s is the instance name now.

Reply:
instances is a plural. It means type already. I updated ling 183 to "Start updating controller type %s"